app/controllers/transcriptions_controller.rb

```diff
@@ -2,19 +2,29 @@ class TranscriptionsController < ApplicationController
   include JSONAPI::Deserialization
 
   def index
-    @transcriptions = Transcription.all
+    @transcriptions = policy_scope(Transcription)
     jsonapi_render(@transcriptions, allowed_filters)
   end
 
-  def update
+  def show
     @transcription = Transcription.find(params[:id])
-    raise ActionController::BadRequest if type_invalid?
-    @transcription.update!(update_attrs)
+    authorize @transcription
     render jsonapi: @transcription
   end
 
-  def show
+  def update
     @transcription = Transcription.find(params[:id])
+    authorize @transcription
+    raise ActionController::BadRequest if type_invalid?
+
+    if approve?
```

Comment:
`:approve` is not a controller action, "approval" just means setting a specific status enum. `approve?` is a boolean method that checks to see if the user is trying to set the transcription status to "approved". I think breaking it up into separate actions would be confusing from the front end's perspective.

I'm not sure what the DSL looks like for reusing policy objects or some such, the `authorize` call itself returns the transcription (if allowed). I could skip the `authorize` call completely and just build the policy and ask it `if policy.update?`, but that skips a bit of the magic and I'm not sure how many cycles we're saving.

Reply:
I don't think you should modify the controller actions at all.

Unless i'm misreading the pundit docs, rather than running multiple pundit authorization actions in the one controller action you can setup your punit policy permission methods to reflect on each other.  E.g`:approve` and 'update` would be pundit policy permission methods, you can specify the pundit permission method you want to check like `authorize(@transcription, :approve?)`

Then your TranscriptionPolicy could look something like
```
class TranscriptionPolicy < ApplicationPolicy
  # ....

  def update?
    has_update_rights?
  end 

  def approve?
    if has_update_rights?
      approver? || admin?
    else
      false
  end

  private

  def has_update_rights?
    admin? || (logged_in? && editor?)
  end
end
```
And your controller action here could be like and you only run the authorization check once and cleanup your controller.
```
  @transcription = Transcription.find(params[:id])
  raise ActionController::BadRequest if type_invalid?
  permission_action =  if approve? 
    :approve?
  else 
    :update?
  end
 authorize @transcription, permission_action
 # ...
```

Reply:
Ah! I get it, great idea! Thought you meant controller action. Added a new policy action and now I'm conditionally calling either `authorize @transcription` or `authorize @transcription, :approve?`. 

A+ refactor thx